### DISCUSSION TITLE
상호작용 중 캐릭터 위치, 동작 종료

### DISCUSSION THREAD

**Discussion by su0833** [naverz/zepeto-studio-kor#547] [5. Q&A]
안녕하세요 제페토 상호작용 만들다 어려운점이 생겨 질문드립니다
상호작용 버튼을 만들고 동작 취하는거 까진 했는데, 버튼을 누르면 캐릭터가 원하는 위치에서 동작을 시작하는것과
이동키를 누르면 애니메이션 동작을 멈추게 하고싶은데 어떻게 스크립트를 짜야 할까요??
자세한 설명 부탁드리겠습니다

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 상호작용 버튼 코드를 어떻게 짰는지 올려주시면 더 상세하게 안내가 가능할 것 같습니다.
원하신 기능 위주로 설명 드립니다.

1. 캐릭터가 원하는 위치로 텔레포트 하는 방법
- Teleport 사용
- 예: this.zepetoCharacter.Teleport(new Vector3(0, 0, 0), Quaternion.identity);
- 가이드 : https://studio.zepeto.me/kr/guides/character-control

2. 이동키를 눌렀을 때를 판별하는 방법 
- ZepetoCharacter.tryMove (boolean flag 사용)
- 예: if (ZepetoCharacter.tryMove)

**Reply by su0833** [reply]
답변 감사합니다^^ 상호작용 버튼 코드는 이렇게 짰습니다
![01](https://user-images.githubusercontent.com/100546192/160786661-77041899-0b35-4ac4-bec1-94e8bb4808c5.png)

**Reply by su0833** [reply]
혹시 위에 코드에 어떻게 적용해야 하는지 알려주실 수 있나요??
혼자서 막 해보고는 있는데 안되네요ㅠㅠ

**Reply by Zepeto-TechEvangelist** [reply]
실제로 사용자가 눌러야 하는 버튼이 interactionButton라고 가정하고 안내 드립니다.

<img width="841" alt="스크린샷 2022-03-31 오후 3 08 34" src="https://user-images.githubusercontent.com/100118650/160990232-ed836650-0cd2-4a64-b57a-92f744791af1.png">

텔레포트 내의 좌표는 원하는대로 수정하세요.
그리고 텔레포트 후 일정시간 후 제스쳐를 취하고
이동키를 입력하면 제스쳐가 캔슬됩니다.

<img width="636" alt="스크린샷 2022-03-31 오후 3 08 47" src="https://user-images.githubusercontent.com/100118650/160990250-ee5b54c3-45f3-4ff6-8915-8fe4b1414fa2.png">

**Reply by su0833** [reply]
![03](https://user-images.githubusercontent.com/100546192/161000013-569c5570-a6be-4a57-a99e-e7c67177eb87.png)

![02](https://user-images.githubusercontent.com/100546192/161000073-72253634-cbef-4957-87e8-0a56238e1110.png)


↑↑ 저는 이렇게 뜨는데 왜이러는 걸까요??

↓↓ 빨간줄 생기는 부분을 바꾸면 이렇게 다른 부분에 빨간줄이 갑니다

![04](https://user-images.githubusercontent.com/100546192/161000270-9927963a-8d2d-4d48-9f30-f173d3cd1db1.png)

**Reply by Zepeto-TechEvangelist** [reply]
Start 부분에 제페토 캐릭터와 변수를 선언해 주고, OnClick.AddListener() 가 중복된 부분을 삭제해주세요.

**Reply by heavybowgun** [reply]
> 

저도 똑같은 문제를 겪고있는데 혹시 Start 부분에 제페토캐릭터와 변수 선언 코드를 알 수 있을까요. 아직 초보라서 보고 따라하고있는데 찾아봐도 잘 모르겠네요.

**Reply by Zepeto-TechEvangelist** [reply]
안녕하세요, 공지 사항에 올라가 있는 샘플 월드 중 인터랙션 부분을 참고해 주시면 도움이 될 것 같습니다.
https://github.com/naverz/zepeto-studio-kor/discussions/615

먼저 프로젝트를 다운받고 실행해보신 후 구조를 이해해보시면 좋습니다. (정말 심플하게 되어 있어서 어렵지 않아요!)
전체 코드는 다음과 같습니다.
https://github.com/naverz/zepeto-world-sample/blob/main/Assets/Chapter3/Scripts/Interaction.ts

감사합니다.
